src/Pardiso.jl

```diff
@@ -128,6 +131,15 @@ function __init__()
         @warn "MKLROOT not set, MKL Pardiso solver will not be functional"
     end
 
+    if mkl_is_available() 
+        try
+            libmklpardiso = Libdl.dlopen(libmkl_rt[])
```

Comment:
Should this make `mkl_is_available()` return false after it has failed to load it?

Reply:
Basically, this test triggers an error when `using Pardiso`, after a second `using Pardiso` one nevertheless one can instantiate `MKLPardisoSolver()`. So yes, seems to be a good idea.